settings: Cluster settings now have page anchors

nickvigilante opened the pull request:
Currently, it's not possible to link to a specific cluster setting within the Cluster Settings page. This makes it difficult for our TSE team to share individual cluster settings and makes it hard for our customers to navigate the page. This PR fixes that.

This PR also addresses a display issue with the cluster setting descriptions not properly escaping the HTML and showing up as malformed in the docs.

Fixes #86400

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/93200)
<!-- Reviewable:end -->

Commit:
settings: Cluster settings now have page anchors

Currently, it's not possible to link to a specific cluster setting
within the Cluster Settings page. This makes it difficult for our TSE
team to share individual cluster settings and makes it hard for our
customers to navigate the page. This PR fixes that.

Fixes #86400

Release note: None

rail reviewed (approved):
LGTM, :shipit:

nickvigilante commented:
bors r+
